discovery/db_discovery.go

```diff
@@ -0,0 +1,100 @@
+package discovery
+
+import (
+	"time"
+
+	"github.com/livepeer/go-livepeer/common"
+	"github.com/livepeer/go-livepeer/core"
+	lpTypes "github.com/livepeer/go-livepeer/eth/types"
+	"github.com/livepeer/go-livepeer/net"
+
+	"github.com/golang/glog"
+)
+
+var cacheRefreshInterval = 1 * time.Hour
+var getTicker = func() *time.Ticker {
+	return time.NewTicker(cacheRefreshInterval)
+}
+
+type DBOrchestratorPoolCache struct {
+	node *core.LivepeerNode
+}
+
+func NewDBOrchestratorPoolCache(node *core.LivepeerNode) *DBOrchestratorPoolCache {
+	if node.Eth == nil {
+		glog.Error("Could not refresh DB list of orchestrators: LivepeerNode nil")
+		return nil
+	}
+
+	ticker := getTicker()
+	go func(node *core.LivepeerNode) {
+		for {
+			select {
+			case <-ticker.C:
+				orchestrators, err := node.Eth.RegisteredTranscoders()
+				if err != nil {
+					glog.Error("Could not refresh DB list of orchestrators: ", err)
+					return
```

Comment:
Probably best to `continue` here; the issue may have resolved itself by the next tick